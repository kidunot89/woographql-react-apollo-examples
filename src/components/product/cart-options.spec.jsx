import React from 'react';
import '@testing-library/react/dont-cleanup-after-each';
import { render, cleanup } from '@testing-library/react';

import { mockProduct } from '../../testing/mocks';
import ApolloMockProvider from '../../testing/apollo-mock-provider';
import SessionProvider from '../session-provider';
import CartOptions from './cart-options';

describe('CartOptions for mocked product', () => {
  afterAll(() => cleanup());
  const { baseElement } = render(
    <ApolloMockProvider>
      <SessionProvider>
        <CartOptions product={mockProduct} />
      </SessionProvider>
    </ApolloMockProvider>,
  );

  it('should render successfully', () => {
    expect(baseElement).toMatchSnapshot();
  });
});
