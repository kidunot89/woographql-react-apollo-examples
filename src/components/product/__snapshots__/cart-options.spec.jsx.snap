// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartOptions for mocked product should render successfully 1`] = `
<body>
  <div>
    <form>
      <div>
        <div>
          <div
            class="font-bold"
          />
          <div>
            <label>
              <input
                checked=""
                class="mr-3 text-center"
                name="Print"
                type="radio"
                value="ankara"
              />
              Ankara
            </label>
          </div>
          <div>
            <label>
              <input
                class="mr-3 text-center"
                name="Print"
                type="radio"
                value="candy-skull"
              />
              Candy skull
            </label>
          </div>
        </div>
      </div>
      <div
        class="flex flex-wrap items-center p-0 w-full mt-5 gap-y-3 lg:flex-nowrap lg:gap-x-3"
      >
        <span
          class="w-1/4 lg:"
        >
          $
          10
        </span>
        <div
          class="inline-flex gap-x-3 items-center w-3/4 justify-end lg:w-1/3 lg:justify-start"
        >
          <span
            class="font-sans font-bold"
          >
            1
          </span>
          <div
            class="relative inline-block w-20"
          >
            <div
              class="appearance-none block text-center leading-5 m-0 py-2.5 px-0 h-8 outline-transparent outline outline-2 outline-offset-2 border rounded quantity-input"
              readonly=""
              type="number"
              value="1"
            />
            <div
              class="icon-button absolute z-10 left-2.5 top-1/2 -translate-y-1/2"
              role="button"
              tabindex="0"
            >
              <i
                class="fa-solid fa-minus"
              />
            </div>
            <div
              class="icon-button absolute z-10 right-2.5 top-1/2 -translate-y-1/2"
              role="button"
              tabindex="0"
            >
              <i
                class="fa-solid fa-plus"
              />
            </div>
          </div>
        </div>
        <button
          class="primary-button w-full disabled:opacity-25 disabled:cursor-not-allowed disabled:text-white lg:1/3"
          type="submit"
        >
          Add To Cart
        </button>
      </div>
    </form>
  </div>
</body>
`;
